### DISCUSSION TITLE
TypeScript support

### DISCUSSION THREAD

**Discussion by rossng** [pietrop/slate-transcript-editor#58] [General]
Hey Pietro! Would you mind if I added TypeScript support to this project? (I know some people are morally opposed to it so didn't want to wade straight in with a PR!)

**Comment by pietrop** [comment]
Hi @rossng 
I don't have much experience with typescript myself, but I have been considering it for this project, as I think it might be at the right stage where adding typescript could make it more robust, easier to maintain and troubleshoot.

I'd welcome a PR, and I think it could be good if it was one of those setup, where it is still not super strict but more of an opt in, if that makes sense?

**Reply by rossng** [reply]
Great 🙂. There are various options in the compiler that make it more or less strict, so I'll try and calibrate accordingly - and make sure that you can still write pure JS files if you want to.

**Reply by pietrop** [reply]
sounds great, thanks!
